### DISCUSSION TITLE
Recovering last returned value

### DISCUSSION THREAD

**Discussion by samoylovfp** [nushell/nushell#11798] [Ideas]
When working with python interpreter, a feature I find useful is that the interpreter saves the last returned value to a variable `_`, so you can do something like
```python
>>> list_resources()  # display output of an expensive call
>>> resources = _     # save the result in a variable that can now be examined, saved to file, or further processed
```

I keep running into a situation when using `nu` where I receive the result of a call that took ~20 seconds just to realize that I forgot to add `let var =` in the beginning.

It would be very nice if there would be a way to get the latest returned value.

Obvious issue here is when the value is large, saving it might take considerable RAM size, but I think this can be easily solved by a "last_value_to_save_max_size_bytes" configuration value, that skips saving values that take more RAM than the value.

How does this sound? Is there already a way to recover the last provided value?

**Comment by fdncred** [comment]
We've talked about this before and our assessment is the same as yours.
1. We'd like to have such a thing.
2. We have to limit it because we support infinite streams.

We don't have a variable we'd like to store it in off-hand and we don't want to be too verbose in naming the config key.

**Reply by samoylovfp** [reply]
Would submitting a PR be a good next step to discuss the details?

**Reply by fdncred** [reply]
The best next step would be discussing how you'd like to implement it in our #design-discussion channel on Discord.
